x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_requirements_page/fleet_server_requirement_page.tsx

```diff
@@ -16,62 +16,231 @@ import {
   EuiText,
   EuiLink,
   EuiEmptyPrompt,
+  EuiSteps,
+  EuiCodeBlock,
+  EuiCallOut,
+  EuiSelect,
 } from '@elastic/eui';
 import styled from 'styled-components';
-import { FormattedMessage } from 'react-intl';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
 
-import { useStartServices } from '../../../hooks';
+import { DownloadStep } from '../components/agent_enrollment_flyout/steps';
+import { useStartServices, useGetOutputs, sendGenerateServiceToken } from '../../../hooks';
+
+const FlexItemWithMinWidth = styled(EuiFlexItem)`
+  min-width: 0px;
+  max-width: 100%;
+`;
 
 export const ContentWrapper = styled(EuiFlexGroup)`
   height: 100%;
+  margin: 0 auto;
+  max-width: 800px;
 `;
 
-function renderOnPremInstructions() {
+// Otherwise the copy button is over the text
+const CommandCode = styled.pre({
+  overflow: 'scroll',
+});
+
+type PLATFORM_TYPE = 'linux-mac' | 'windows' | 'deb-rpm';
+const PLATFORM_OPTIONS: Array<{ text: string; value: PLATFORM_TYPE }> = [
+  { text: 'Linux / macOS', value: 'linux-mac' },
+  { text: 'Windows', value: 'windows' },
+  { text: 'RPM / DEB', value: 'deb-rpm' },
+];
+
+const OnPremInstructions: React.FC = () => {
+  const outputsRequest = useGetOutputs();
+  const { notifications } = useStartServices();
+  const [serviceToken, setServiceToken] = useState<string>();
+  const [isLoadingServiceToken, setIsLoadingServiceToken] = useState<boolean>(false);
+  const [platform, setPlatform] = useState<PLATFORM_TYPE>('linux-mac');
+
+  const output = outputsRequest.data?.items?.[0];
+  const esHost = output?.hosts?.[0];
+
+  const installCommand = useMemo((): string => {
+    if (!serviceToken || !esHost) {
+      return '';
+    }
+    switch (platform) {
+      case 'linux-mac':
+        return `sudo ./elastic-agent install -f --fleet-server-service-token=${serviceToken} --fleet-server-es=${esHost}`;
+      case 'windows':
+        return `.\\elastic-agent.exe install --fleet-server-service-token=${serviceToken} --fleet-server-es=${esHost}`;
```

Comment:
On windows you will want to run that either in a `cmd.exe` or a Powershell that is started with `Run as Administrator`.